MyLibrary/MyLibraryApp.API/Controllers/BooksController.cs

```diff
@@ -0,0 +1,78 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using MyLibraryApp.Application.Models.Books;
+using MyLibraryApp.Application.Services.Interfaces;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace MyLibraryApp.API.Controllers
+{
+    public class BooksController :ControllerBase
+    {
+        private readonly IBookService bookService;
+        public BooksController(IBookService bookService)
+        {
+            this.bookService = bookService;
+        }
+
+        [HttpGet]
+        public async Task<IActionResult> GetAllBooksAsync()
+        {
+            var response = await bookService.ReadAllBooksAsync();
+
+            return Ok(response);
+        }
+
+        [HttpGet]
+        public async Task<IActionResult> GetBookByIdAsync(int id)
+        {
+            var response = await bookService.GetBookByIdAsync(id);
+
+            return Ok(response);
+        }
+        [HttpGet]
+        public async Task<IActionResult> GetBookByAuthorAsync(string author)
```

Comment:
I assume you get the book by author's name? If yes, the name of the method in of the parameter needs to reflect that. 

A better practice is to use id's as it's usually more efficient to search by id than any other field. Thant would mean you method would look something like `GetBookByAuthorIdAsync(int authorId)`.